src/providers/GlobalContext/GlobalContext.jsx

```diff
@@ -33,12 +26,47 @@ export function GlobalContextProvider({ children }) {
     }
   };
 
+  const favoritesReducer = (state, action) => {
+    switch (action.type) {
+      case 'add':
+        return [...state, action.payload];
+      case 'remove':
+        return state.filter((v) => v.videoId !== action.payload);
+      default:
+        throw new Error('Invalid favoritesReducer action type');
+    }
+  };
+
   const [searchState, searchTextDispatch] = useReducer(searchTextReducer, 'wizeline');
   const [themeState, themeDispatch] = useReducer(themeReducer, 'lightTheme');
+  const [openLogin, setOpenLogin] = useState(false);
+  const initialUser = JSON.parse(sessionStorage.getItem('user'));
+  const [user, setUser] = useState(initialUser);
+  const initialFavorites = JSON.parse(localStorage.getItem('favorites')) || [];
+  const [favorites, favoritesDispatch] = useReducer(favoritesReducer, initialFavorites);
```

Comment:
You should have just one reducer for the whole state of your application, so that you have a single source of truth for it.